pytorch_lightning/trainer/connectors/logger_connector.py

```diff
@@ -172,8 +172,10 @@ def __log_evaluation_epoch_metrics_2(self, eval_results, test_mode):
                 # log metrics
                 self.trainer.logger_connector.log_metrics(log_metrics, {})
 
-                # track metrics for callbacks
+                # track metrics for callbacks (all prog bar, logged and callback metrics)
                 self.trainer.logger_connector.callback_metrics.update(callback_metrics)
+                self.trainer.logger_connector.callback_metrics.update(log_metrics)
```

Comment:
can one distinguish whether they want to send certain values to callbacks vs loggers vs progress bars?
i'm thinking of a use case where we calculate huge embeddings. we'd want the callback to access them to write them out to some storage, but they're not also used in the logger. memory could be a concern here

Reply:
these are only scalars.

this is exactly why you have the result objects... 
.log(prog_bar=True, logger=False) means ONLY progress bar
.log(prog_bar=False, logger=True) means ONLY log
.log(prog_bar=True, logger=True) means both


This update i'm making just gives the content of both to be able to monitor for callbacks. Everything is detached though, so no memory leaks.

For writing embeddings we have the .write in the result. that goes straight to disk.
result.write(x)

Reply:
1. you wouldn't log and embedding.
2. If you wanted a callback to access an embedding, then you'd save it in the pl_module and access it there in the callback.

(like we do in some of the bolts callbacks)